src/polycap-photon.c

```diff
@@ -1018,24 +1018,40 @@ bool polycap_photon_get_extleak_data(polycap_photon *photon, polycap_leak ***lea
 		return false;
 	}
 
-	fprintf(stderr, "C: n_extleak: %lld\n", photon->n_extleak);
 
 	*n_leaks = photon->n_extleak;
+	fprintf(stderr, "C: n_extleak: %lld\n", photon->n_extleak);
+	fprintf(stderr, "C: n_energies: %lld\n", photon->extleak[0]->n_energies);
 	if (photon->n_extleak == 0){
 		*leaks = NULL;
 		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_photon_get_extleak_data: no extleak events in photon");
 		return false;
 	}
+
 	*leaks = malloc(sizeof(polycap_leak*) * photon->n_extleak);
 	if (*leaks == NULL){
 		polycap_set_error(error, POLYCAP_ERROR_MEMORY, "polycap_photon_get_extleak_data: could not allocate memory for leaks -> %s", strerror(errno));
 		return false;
 	}
+
+	fprintf(stderr, "here\n");
 	for(i = 0; i < photon->n_extleak; i++) {
+		fprintf(stderr, "here?1\n");
 		*leaks[i] = malloc(sizeof(polycap_leak));
```

Comment:
@tschoonj Het probleem zit hier...
Wanneer n_extleak > 1, dan geeft hij op deze lijn error:
`Invalid write of size 8`
Je krijgt de error heel snel met `valgrind ./tests/leaks` daar hij niet aan de `polycap_source` tests komt ;)